app/pages/reporter/RoleInArrangementPage.scala

```diff
@@ -16,13 +16,38 @@
 
 package pages.reporter
 
+import models.UserAnswers
 import models.reporter.RoleInArrangement
+import models.reporter.RoleInArrangement.{Intermediary, Taxpayer}
 import pages.QuestionPage
+import pages.reporter.intermediary._
+import pages.reporter.taxpayer._
 import play.api.libs.json.JsPath
 
+import scala.util.Try
+
 case object RoleInArrangementPage extends QuestionPage[RoleInArrangement] {
 
   override def path: JsPath = JsPath \ toString
 
   override def toString: String = "roleInArrangement"
+
+  override def cleanup(value: Option[RoleInArrangement], userAnswers: UserAnswers): Try[UserAnswers] = {
+
+    value match {
+      case Some(Intermediary) =>
+        userAnswers.remove(TaxpayerWhyReportInUKPage)
```

Comment:
Optional: why not a list + foldLeft?

Reply:
no type parameters for method remove: (page: pages.QuestionPage[A])scala.util.Try[models.UserAnswers] exist so that it can be applied to arguments (pages.QuestionPage[_ >: java.time.LocalDate with models.reporter.taxpayer.TaxpayerWhyReportArrangement with models.reporter.taxpayer.TaxpayerWhyReportInUK <: Object] with Product with Serializable)
 --- because ---
argument expression's type is not compatible with formal parameter type;
 found   : pages.QuestionPage[_ >: java.time.LocalDate with models.reporter.taxpayer.TaxpayerWhyReportArrangement with models.reporter.taxpayer.TaxpayerWhyReportInUK <: Object] with Product with Serializable
 required: pages.QuestionPage[?A]
 
 Keep getting this error message? if I use the foldLeft any ideas how it resolve?